include/waflz/def.h

```diff
@@ -72,20 +79,28 @@
 //: ----------------------------------------------------------------------------
 //: types
 //: ----------------------------------------------------------------------------
+#ifdef __cplusplus
 namespace ns_waflz {
 typedef enum {
-	PART_MK_ACL = 1,
-	PART_MK_WAF = 2,
+        PART_MK_ACL = 1,
+        PART_MK_WAF = 2,
         PART_MK_RULES = 4,
         PART_MK_LIMITS = 8,
-	PART_MK_ALL = 15,
+        PART_MK_ALL = 15,
 } part_mk_t;
+#endif
+//: ----------------------------------------------------------------------------
+//: constants
+//: ----------------------------------------------------------------------------
+#define DEFAULT_BODY_SIZE_MAX (128*1024)
 // callbacks
-typedef int32_t (*get_rqst_data_size_cb_t)(uint32_t &, void *);
-typedef int32_t (*get_rqst_data_cb_t)(const char **, uint32_t &, void *);
-typedef int32_t (*get_rqst_data_w_key_cb_t)(const char **, uint32_t &, void *, const char *, uint32_t);
-typedef int32_t (*get_rqst_kv_w_idx_cb_t)(const char **, uint32_t &, const char **, uint32_t &, void *, uint32_t);
-typedef int32_t (*get_rqst_body_data_cb_t)(char *, uint32_t &, bool &, void *, uint32_t);
+typedef int32_t (*get_rqst_data_size_cb_t)(uint32_t *a, void *);
+typedef int32_t (*get_rqst_data_cb_t)(const char **, uint32_t *, void *);
+typedef int32_t (*get_rqst_data_w_key_cb_t)(const char **, uint32_t *, void *, const char *, uint32_t);
+typedef int32_t (*get_rqst_kv_w_idx_cb_t)(const char **, uint32_t *, const char **, uint32_t *, void *, uint32_t);
+typedef int32_t (*get_rqst_body_data_cb_t)(char *, uint32_t *, bool , void *, uint32_t *);
```

Comment:
fourth parameter is pointer intentionally or typo?

Reply:
intentionally made it to a pointer instead of pass by value. Could have left it as it is.